genshin/errors.py

```diff
@@ -107,6 +107,30 @@ class AuthkeyTimeout(AuthkeyException):
 
     msg = "Authkey has timed out."
 
+class RedemptionExpired(GenshinException):
+    """ The redemption has expired """
+
+    msg = "The redemption has expired."
+
+class RedemptionInvalid(GenshinException):
+    """ Invalid redemption code. """
+
+    msg = "Invalid redemption code."
+
+class RedemptionCooldown(GenshinException):
+    """ Cooldown for redeeming """
+
+    msg = "Redemption is on cooldown."
+
+class RedemptionClaimed(GenshinException):
+    """ Redeption code has been claimed already. """
+
+    msg = "Redeption code has been claimed already."
+
+class RedemptionARLower(GenshinException):
+    """ Cannot claim codes for account with adventure rank lower than 10. """
+
+    msg = "Cannot claim codes for account with adventure rank lower than 10."
```

Comment:
surely these could all be just one class right?

Reply:
Yes

Reply:
I use for Discord bot. because I want to error handle like this.
![image](https://user-images.githubusercontent.com/24217516/152570392-87d45e2a-26bd-4419-88a6-8a25067badeb.png)


Sorry for english is not very well

Reply:
ok, that's fine. Some errors can be joined into a single class though.